src/main/java/com/limechain/runtime/hostapi/dto/VerifySignature.java

```diff
@@ -0,0 +1,44 @@
+package com.limechain.runtime.hostapi.dto;
+
+import lombok.Getter;
+import lombok.Setter;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+@Getter
+@Setter
+public class VerifySignature {
+
+    private byte[] signatureData;
+    private byte[] messageData;
+    private byte[] publicKeyData;
+
+    private final Key key;
+
+    public VerifySignature(byte[] signatureData, byte[] messageData, byte[] publicKeyData, Key key) {
+        this.signatureData = signatureData;
+        this.messageData = messageData;
+        this.publicKeyData = publicKeyData;
+        this.key = key;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        VerifySignature verifySig = (VerifySignature) o;
+        return Arrays.equals(signatureData, verifySig.signatureData) &&
+                Arrays.equals(messageData, verifySig.messageData) &&
+                Arrays.equals(publicKeyData, verifySig.publicKeyData) && key == verifySig.key;
+    }
+
+    @Override
+    public int hashCode() {
+        int result = Objects.hash(key);
+        result = 31 * result + Arrays.hashCode(signatureData);
```

Comment:
Autogenerated hashcode function.

A nonzero initial value is used so the hash value will be affected by initial fields whose hash value is zero.

However, in Effective Java, it is suggested to use prime numbers in which 31 is preferred for faster calculations:
A nice property of 31 is that the multiplication can be replaced by a shift and a subtraction for better performance: 31 * i == (i << 5) - i. Modern VMs do this sort of optimization automatically.

The multiplier 31 (used to be 37, but as discussed earlier 31 is better for performance) was chosen because it is an odd prime. If it was even and the multiplication overflowed, information would be lost because multiplication by two is equivalent to shifting. The advantages of using a prime number are less clear, but it is traditional to use primes for this purpose.